src/components/menu/menu.tsx

```diff
@@ -150,28 +176,181 @@ export class Menu {
-                            }}
-                            items={this.items}
-                            type="menu"
-                            badgeIcons={this.badgeIcons}
-                            onSelect={this.handleSelect}
-                            ref={this.setListElement}
-                        />
+                        {this.renderBreadcrumb()}
+                        {this.renderMenuList()}
                     </limel-menu-surface>
                 </limel-portal>
             </div>
         );
     }
 
-    public componentDidRender() {
-        const slotElement = this.host.shadowRoot.querySelector('slot');
-        slotElement.assignedElements().forEach(this.setTriggerAttributes);
+    @Watch('items')
+    protected itemsWatcher() {
+        this.currentItems = null;
+
+        this.setFocus();
+    }
+
+    @Watch('currentItems')
+    protected currentItemsWatcher() {
+        this.setFocus();
     }
 
+    @Watch('open')
+    protected openWatcher() {
+        if (!this.open) {
+            this.currentItems = null;
+
+            return;
+        }
+
+        this.setFocus();
+    }
+
+    private renderBreadcrumb = () => {
+        if (!this.currentSubMenu) {
+            return;
+        }
+
+        const breadCrumbItems: MenuCrumbItem[] = [];
+        let currentItem = this.currentSubMenu;
+        while (currentItem) {
+            breadCrumbItems.unshift({
+                text: currentItem.text,
+                icon: {
+                    name: currentItem.icon,
+                    color: currentItem.iconColor,
+                },
+                menuItem: currentItem,
+            });
+            currentItem = currentItem.parentItem;
+        }
+
+        if (!breadCrumbItems.length) {
+            return;
+        }
+
+        return (
+            <limel-breadcrumbs
+                style={{
+                    'border-bottom': 'solid 1px rgb(var(--contrast-500))',
+                    'flex-shrink': '0',
+                }}
+                onSelect={(
+                    event: LimelBreadcrumbsCustomEvent<MenuCrumbItem>
+                ) => {
+                    if (!event.detail.menuItem) {
+                        this.currentItems = null;
+                        this.currentSubMenu = null;
+                        this.navigateMenu.emit(null);
+
+                        return;
+                    }
+
+                    this.handleSelect(event.detail.menuItem);
+                }}
+                items={[
+                    {
+                        text: '',
+                        icon: {
+                            name: 'home',
+                        },
+                        type: 'icon-only',
+                    },
+                    ...breadCrumbItems,
+                ]}
+            />
+        );
+    };
+
+    private renderMenuList = () => {
+        const items = this.visibleItems;
+
+        return (
+            <limel-menu-list
+                tabindex="1"
```

Comment:
> The failing test seems to be failing because of this tabindex. (Although, I tried running the tests locally, both with this left in, and with it removed, and all tests passed regardless 🤷‍♂️)

Ah, no, I tried re-running the tests now, and they passed, so it was just a timing-thing or something I guess. The tests can be a little bit flaky sometimes… 😒

I'd still like to know what this does, or remove it 🙂

Reply:
We can remove it for now. It might be needed once search is added in a later PR.